tools/affiliations/importer/src/main.rs

```diff
@@ -1,17 +1,399 @@
 
+#[macro_use]
+extern crate serde;
+extern crate serde_json;
+
 #[macro_use]
 extern crate dotenv_codegen;
 
+#[macro_use]
+extern crate log;
+
+use std::collections::HashSet;
+
+use chrono::Utc;
 use crawler::crawler::{Crawl, Crawler};
 
+/// Invalid affiliations.
+static INVALID_AFFILIATIONS: [&str; 4] = ["", "-", "(Unknown)", "NotFound"];
+/// PingCAP affiliation.
+static PINGCAP_AFFILIATION: &str = "PingCAP";
+/// Independent affiliation.
+static INDEPENDENT_AFFILIATION: &str = "Independent";
+/// Company separator.
+static COMPANY_SEPARATOR: &str = ", ";
+/// Date separator.
+/// For example: PingCAP < 2021-06-04.
+static DATE_SEPARATOR: &str = " < ";
+
+/// GitHub user info.
+#[derive(Debug, Serialize, Deserialize)]
+struct GitHubUser {
+    login: String,
+    email: String,
+    affiliation: Option<String>,
+    source: Option<String>,
+    name: Option<String>,
+    commits: i64,
+    location: Option<String>,
+    country_id: Option<String>,
+    modified_by_affiliations: Option<bool>,
+}
+
 #[tokio::main]
 async fn main() {
     pretty_env_logger::init();
+
+    info!("Start downloading the github_users file...");
+    let resp = reqwest::get(dotenv!("GITHUB_USERS_SOURCE"))
+        .await
+        .expect("Failed to download the github_users file.")
+        .text()
+        .await
+        .expect("Failed to convert github_users to text.");
+    info!("Downloading the github_users file was successful!");
+
+    let mut records: Vec<GitHubUser> =
+        serde_json::from_str(resp.as_str()).expect("Failed to parse github_users file.");
+    info!("{} records were obtained.", records.len());
+
+    info!("Start fetching the PingCAP GitHub logins...");
     let crawler =
         &Crawler::new(dotenv!("APP_ID"), dotenv!("APP_SECRET")).expect("Failed init the crawler.");
-    match crawler.list_github_logins().await {
-        Ok(names) => names.iter().for_each(|n| println!("{}", n)),
-        Err(error) => panic!("{:#?}", error),
+    let logins = crawler
+        .list_github_logins()
+        .await
+        .expect("Failed to list logins.");
+    let logins: HashSet<String> = logins
+        .iter()
+        .map(|l| l.trim())
+        // NOTICE: login is not case-sensitive.
+        .map(|l| l.to_lowercase())
+        .collect::<HashSet<_>>();
+    info!("Fetching the logins to succeed!");
+
+    // Because a login may appear twice in the original data and their affiliation may be different,
+    // we need to process both, not just one and then consider the process complete.
+    let mut processed_logins: HashSet<String> = HashSet::new();
+    for record in &mut records {
+        // NOTICE: login is not case-sensitive.
+        let login = record.login.to_lowercase();
+        if logins.contains(&login) {
+            set_pingcap_affiliation(record);
+            processed_logins.insert(login);
+        } else {
+            remove_pingcap_affiliation(record);
+        }
+    }
+
+    // Processes the remaining logins.
```

Comment:
```suggestion
    // Processes the remaining logins that existed in the logins but not in the github_logins file.
```

Reply:
Here are the rest of the logins in github_loings.